docs/cookbook/pii_tutorial.md

```diff
@@ -0,0 +1,207 @@
+]
+
+tickets_cleaned = []
+
+for ticket_text in tickets:
+    # Clean the PII from the current ticket
+    ticket_text_cleaned = eunomia.run(ticket_text)
+
+    # Append the sanitized ticket to the cleaned list
+    tickets_cleaned.append(ticket_text_cleaned)
+
+for i, cleaned_ticket in enumerate(tickets_cleaned, start=1):
+    print(f"Cleaned Ticket {i}:\n{cleaned_ticket}\n")
+```
+
+---
+
+## Secure Retrieval of User Information in Chatbots
+
+### ID-Based Access Control: Customizing Data Accessibility
+
+Eunomia’s ID-Based Access Control (IDBAC) provides precise control over which information is accessible based on the identity of the requestor.
+
+#### Key Components of IDBAC
+
+1. **ID-Based Mechanism**: Each user or entity is assigned a unique identifier.
+2. **Document Ownership**: Every document is linked to the unique ID of its owner.
+3. **Global PII Masking**: Specific types of PII are consistently masked, regardless of the viewer.
+4. **Owner-Specific PII Access**: Owners can view their own PII while it remains masked for others.
+
+#### Step 1: Setup
+
+```python
+from eunomia.orchestra import Orchestra
+from eunomia.instruments import PiiInstrument, IdbacInstrument
+```
+
+#### Step 2: Configuring Eunomia
+
+```python
+eunomia = Orchestra(
+    instruments=[
+        IdbacInstrument(
+            instruments=[
+                PiiInstrument(entities=["PERSON", "EMAIL_ADDRESS", "PHONE_NUMBER"], edit_mode="replace")
+            ]
+        ),
+    ]
+)
+```
+
+#### Step 3: Implementing the Final Solution
+
+```python
+tickets = [
+    {
+        "doc_id": 12345,
+        "text": """Ticket ID: #IT-2024-0001
+        User Information:
+        Name: John Doe
+        Email: john.doe@example.com
+        Phone Number: +1 (555) 123-4567
+        Issue Description: Cannot connect to the company's VPN."""
+    },
+    {
+        "doc_id": 98765,
+        "text": """Ticket ID: #IT-2024-0002
+        User Information:
+        Name: Jane Smith
+        Email: jane.smith@example.com
+        Phone Number: +1 (555) 987-6543
+        Issue Description: Application crashes on startup."""
+    }
+]
+
+user_requests = [
+    {"user_id": 12345, "doc_id": 12345},  # Matching user and document ID
+    {"user_id": 12345, "doc_id": 98765}   # Non-matching user and document ID
+]
+
+for request in user_requests:
+    user_id = request["user_id"]
+    doc_id = request["doc_id"]
+
+    # Retrieve and process the ticket text based on IDBAC rules
+    for ticket in tickets:
+        if ticket["doc_id"] == doc_id:
+            processed_text = eunomia.run(ticket["text"], user_id=user_id, doc_id=doc_id)
+
+            # Output the results
+            access_status = "Matching IDs" if user_id == doc_id else "Non-Matching IDs"
+            print(f"Access Status: {access_status}")
+            print(f"User ID: {user_id}, Document ID: {doc_id}")
+            print(f"Processed Ticket:\n{processed_text}\n")
+
+```
```

Comment:
In general I think cookbook's code blocks should contain outputs as well. Can we add the output of `print` functions?